pilot/pkg/model/context.go

```diff
@@ -87,6 +87,13 @@ type Proxy struct {
 	// namespace.
 	ID string
 
+	// Locality is the location of where proxy runs.
+	// For example:
+	// 'region1/zone1/subzone1' means complete region zone and subzone info.
+	// 'region1/zone1' means only region and zone info.
+	// 'region1' means only region info.
+	Locality string
```

Comment:
This is not going to work. We tried doing this a while ago. The problem is that we dont know the locality of the proxy until we query the API server (either from Pilot or from PilotAgent). In the past, we had the kubeClient built into PilotAgent such that on startup, it would query the api server, find out its locality and then start Envoy with the appropriate settings. The problem was that the kubeClient was very heavy, increasing the memory usage of every sidecar by 40-60Mb. Also, it would cause startup delays until the agent obtained the proxy's locality.

So we decided to remove the locality thing from the Pilot agent.

Reply:
I just read the rest of the PR. ignore the comment above

Reply:
Thanks for this context.